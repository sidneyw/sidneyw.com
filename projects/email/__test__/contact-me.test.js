const contact = require('../contact-me');
const SimpleMailer = require('../modules/simple-mail');

describe('Contact Form', () => {
  let mailClient;
  let mockSend;
  let event;
  let sender;
  let recipient;

  beforeEach(() => {
    event = {
      httpMethod: 'POST',
      body: {
        email: 'testemail@mail.com',
        message: "Let's work together",
        name: 'Test User'
      },
      headers: {
        'content-type': 'application/json'
      }
    };

    sender = 'sidney.sender@test.com';
    recipient = 'sidney.recipient@test.com';

    mockSend = jest.fn();
    mockSend.mockReturnValue({ MessageId: 'test-msg-id' });

    mailClient = new SimpleMailer({
      async sendEmail(params) {
        return mockSend(params);
      }
    });
  });

  test('Sends an email', async () => {
    const { statusCode, body } = await contact({ mailClient, sender, recipient }, event);

    expect(statusCode).toBe(200);
    expect(body).toBe('Contact Form Recieved');
    expect(mockSend.mock.calls).toMatchSnapshot();
  });

  test('Errors if event is missing field', async () => {
    delete event.body.name;

    const { statusCode } = await contact({ mailClient, sender, recipient }, event);
    expect(statusCode).toBe(400);
  });

  test('Returns 500 if email fails', async () => {
    mailClient = new SimpleMailer({
      async sendEmail() {
        throw new Error('Oops');
      }
    });

    const { statusCode, body } = await contact({ mailClient, sender, recipient }, event);
    expect(statusCode).toBe(500);
    expect(body).toBe('Internal Server Error');
  });
});
