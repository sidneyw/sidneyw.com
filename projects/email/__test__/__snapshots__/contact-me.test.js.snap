// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form Sends an email 1`] = `
Array [
  Array [
    Object {
      "Destination": Object {
        "ToAddresses": Array [
          "sidney.recipient@test.com",
        ],
      },
      "Message": Object {
        "Body": Object {
          "Html": Object {
            "Charset": "UTF-8",
            "Data": "Let's work together",
          },
          "Text": Object {
            "Charset": "UTF-8",
            "Data": "Let's work together",
          },
        },
        "Subject": Object {
          "Charset": "UTF-8",
          "Data": "sidneyw.com - Contact Form",
        },
      },
      "Source": "Test User <sidney.sender@test.com>",
    },
  ],
]
`;
